sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -100,10 +100,8 @@ public RetryPolicy(RetryStrategy retryStrategy) {
      * @throws NullPointerException If {@code retryOptions} is null.
      */
     public RetryPolicy(RetryOptions retryOptions) {
-        this(
-            getRetryStrategyFromOptions(
-                Objects.requireNonNull(retryOptions, "'retryOptions' cannot be null.")),
-            null, null);
+        this(getRetryStrategyFromOptions(Objects.requireNonNull(retryOptions, "'retryOptions' cannot be null.")), null,
```

Comment:
nit: a lot going on in this call to ctor overload. Maybe we can move the null check down to the `getRetry...` method to make this more readable.